src/App.tsx

```diff
@@ -1,10 +1,56 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { AddNewTodo } from './components/AddNewTodo/AddNewTodo';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterQuery, setFilterQuery] = useState('all');
+
+  // LOAD TODOs
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  // FILTER TODOs
+  const filteredTodos = useMemo(() => {
+    switch (filterQuery) {
+      case 'active': {
+        return todos.filter(todo => !todo.completed);
+      }
+
+      case 'completed': {
+        return todos.filter(todo => todo.completed);
+      }
+
+      default: {
+        return todos;
+      }
+    }
+  }, [filterQuery, todos]);
+
+  // COUNT TODOs
+  const activeTodoCount = useMemo(() => {
+    return [...todos].filter(todo => !todo.completed).length;
```

Comment:
it's redundant to copy the array, the filter method doesn't mutate an array

Reply:
Yeap. Thanks! My bad)